tests/snapshot/snapshot.go

```diff
@@ -78,9 +83,23 @@ func snapshotRestore(t *testing.T, client *rancher.Client, clusterName, poolName
 		},
 	}
 
-	deploymentResp, err := deploy.CreateDeployment(steveclient, initialWorkloadName, deployment)
+	deploymentResp, err := steveclient.SteveType(DeploymentSteveType).Create(deployment)
 	require.NoError(t, err)
 
+	err = kwait.PollUntilContextTimeout(context.TODO(), timeouts.FiveSecondTimeout, timeouts.FiveMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
Left a similar comment below, but I think this makes sense to have abstracted. It's a bit redundant and you can pass in the `deploymentResp` and `postDeploymentResp` as a parameter.